mars/tensor/datasource/array.py

```diff
@@ -151,9 +152,18 @@ def tensor(data, dtype=None, order='K', chunk_size=None, gpu=None, sparse=False)
         return scalar(data, dtype=dtype)
     elif issparse(data):
         return _from_spmatrix(data, dtype=dtype, chunk_size=chunk_size, gpu=gpu)
+    elif hasattr(data, '__mars_tensor__'):
+        data = data.__mars_tensor__(dtype=dtype)
+        return data.astype(data.dtype, order=order, copy=False)
```

Comment:
Now that we already do ``data.astype`` in ``__mars_tensor__``, we may pass order as an argument as well.

Reply:
Done.